builders/precompiles/xcm-Interface.md

```diff
@@ -0,0 +1,233 @@
+description: Learn about the XCM interface precompile and how to use it to transfer assets from moonbeam networks to other parachains.
+---
+
+# XCM Interface precompile
+
+## Introduction {: #introduction }
+
+As a Polkadot parachain, Moonbeam has the inherent ability to communicate and exchange data with other connected parachains. This native cross-chain communication allows safe and fast token transfers leveraging the Cross-Consensus Message format (XCM for short), facilitating communication between different consensus systems.
+
+The communication protocol enabling token transfers is built on [Substrate](/builders/substrate/){target=\_blank} and runs on a lower level than the EVM, making it harder for EVM developers to access.
+
+Nevertheless, Moonbeam networks have an XCM precompile that fills the gap between execution layers, exposing a smart contract interface that abstracts away the underlying complexities, making the execution of cross-chain token transfers as easy as any other smart contract call.
+
+This guide will show you how to interact with the [XCM interface](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank} precompile to execute cross-chain token transfers through the Ethereum API.
+
+The XCM precompile is located at the following address:
+
+=== "Moonbeam"
+
+     ```text
+     {{networks.moonbeam.precompiles.xcm_interface }}
+     ```
+
+=== "Moonriver"
+
+     ```text
+     {{networks.moonriver.precompiles.xcm_interface }}
+     ```
+
+=== "Moonbase Alpha"
+
+     ```text
+     {{networks.moonbase.precompiles.xcm_interface }}
+     ```
+
+--8<-- 'text/builders/ethereum/precompiles/security.md'
+
+## The XCM Solidity Interface {: #the-xcm-solidity-interface }
+
+The [`XCMInterface.sol`](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank} is a Solidity interface that allows developers to interact with the precompile's functions.
+
+??? code "XcmInterface.sol"
+
+    ```solidity
+    --8<-- 'code/builders/precompiles/xcm-interface/XcmInterface.sol'
+    ```
+
+The interface includes the necessary data structures along with the following functions:
+
+???+ function "**transferAssetsToPara20**(_paraId, beneficiary, assets, feeAssetItem, weight_) — sends assets to another EVM-compatible appchain using the underlying `transfer_assets()` transaction included in the XCM pallet module"
+
+    === "Parameters"
+
+        - `paraId` (uint32) - the destination's appchain ID
+        - `beneficiary` (address) - the ECDSA-type account in the destination chain that will receive the tokens
+        - `assets` (`AssetAddressInfo[]` memory) - an array of assets to send
+        - `feeAssetItem` (uint32) - the index of the asset that will be used to pay fees
+        - `weight` (Weight memory)- the maximum gas to use in the whole operation. Setting uint64::MAX to `refTime` acts in practice as *unlimited weight*
```

Comment:
This is inconsistent with the way we document types on Moonbeam right now. There are some incoming changes, but for now, I think just throw the types into italics (i.e., `paraId` _uint32_, the destination's appchain ID) and that will be a quick win that is close enough